pkg/multitenant/hashing.go

```diff
@@ -0,0 +1,42 @@
+package multitenant
+
+import (
+	"context"
+
+	"github.com/kumahq/kuma/pkg/core/resources/store"
+)
+
+type Hashing interface {
+	GetOptionsFunc() store.GetOptionsFunc
+	ListOptionsFunc() store.ListOptionsFunc
+	KdsHashFn(ctx context.Context, id string) string
+	SinkStatusCacheKey(ctx context.Context) string
```

Comment:
Now that I see it like this is there a real difference between these 3 different hashing methods? Should we just add a suffix or prefix to an already existing hash key? What is the value of having 3 different methods?

Reply:
I thought different implementations might be needed in the parent project but probably not. I'll leave just one.